dnn: build fails on VS2012 with 'round' identifier not found

<!--
If you have a question rather than reporting a bug please go to http://answers.opencv.org where you get much faster responses.
If you need further assistance please read [How To Contribute](https://github.com/opencv/opencv/wiki/How_to_contribute).

This is a template helping you to create an issue which can be processed as quickly as possible. This is the bug reporting section for the OpenCV library.
-->

##### System information (version)
<!-- Example
- OpenCV => 3.1
- Operating System / Platform => Windows 64 Bit
- Compiler => Visual Studio 2015
-->

- OpenCV => master ( after 17dcf0e82dc5f86081f1ae67a5b1ec0412e9d5cc #10255)
- Operating System / Platform => Windows 10 Pro 64bit
- Compiler => Visual Studio 2012

##### Detailed description
 - When building recent OpenCV master on Visual Studio 2012, following error appears.
````
  C:\opencv\modules\dnn\src\layers\pooling_layer.cpp(299): error C3861: 'round': identifier not found [C:\opencv\build\modules\dnn\opencv_dnn.vcxproj]
  C:\opencv\modules\dnn\src\layers\pooling_layer.cpp(300): error C3861: 'round': identifier not found [C:\opencv\build\modules\dnn\opencv_dnn.vcxproj]
  C:\opencv\modules\dnn\src\layers\pooling_layer.cpp(307): error C3861: 'round': identifier not found [C:\opencv\build\modules\dnn\opencv_dnn.vcxproj]
  C:\opencv\modules\dnn\src\layers\pooling_layer.cpp(308): error C3861: 'round': identifier not found [C:\opencv\build\modules\dnn\opencv_dnn.vcxproj]
````
 - The actual place of the error is [here](https://github.com/opencv/opencv/blob/17dcf0e82dc5f86081f1ae67a5b1ec0412e9d5cc/modules/dnn/src/layers/pooling_layer.cpp#L299-L308).
 - The root of the cause is that ```ronud``` function doesn't exists on Visual Studio 2012 ( and probably earlier )
 - I'll throw a PR to fix this issue, later.

##### Steps to reproduce
 - Build OpenCV revision 17dcf0e82dc5f86081f1ae67a5b1ec0412e9d5cc or later on Visual Studio 2012

--- a/modules/dnn/src/layers/pooling_layer.cpp
+++ b/modules/dnn/src/layers/pooling_layer.cpp
@@ -57,6 +57,10 @@ namespace cv
 {
 namespace dnn
 {
+static inline int scaleAndRoundRoi(float f, float scale)
+{
+    return (int)(f * scale + (f >= 0.f ? 0.5f : -0.5f));
+}
 
 class PoolingLayerImpl : public PoolingLayer
 {
@@ -296,16 +300,16 @@ public:
                 if (poolingType == ROI)
                 {
                     const float *roisData = rois->ptr<float>(n);
-                    int ystartROI = round(roisData[2] * spatialScale);
-                    int yendROI = round(roisData[4] * spatialScale);
+                    int ystartROI = scaleAndRoundRoi(roisData[2], spatialScale);
+                    int yendROI = scaleAndRoundRoi(roisData[4], spatialScale);
                     int roiHeight = std::max(yendROI - ystartROI + 1, 1);
                     roiRatio = (float)roiHeight / height;
 
                     ystart = ystartROI + y0 * roiRatio;
                     yend = ystartROI + std::ceil((y0 + 1) * roiRatio);
 
-                    xstartROI = round(roisData[1] * spatialScale);
-                    int xendROI = round(roisData[3] * spatialScale);
+                    xstartROI = scaleAndRoundRoi(roisData[1], spatialScale);
+                    int xendROI = scaleAndRoundRoi(roisData[3], spatialScale);
                     int roiWidth = std::max(xendROI - xstartROI + 1, 1);
                     roiRatio = (float)roiWidth / width;
 

